release/release-source

```diff
@@ -135,101 +136,70 @@ prepare()
-                tar -C $stagedir --exclude="node_modules" \
-                    --strip-components=1 -xf "$archive"
-                git -C $stagedir add -f .
-                git -C $stagedir commit -q --message="initial"
-                git -C $stagedir tag -a initial --message="initial"
-            fi
-
-            trace "Creating patched tarball"
-
-            # Some software looks at its version number from the git tag
-            # so we move the git version tag here to this specific patch commit
-            git -C $repodir tag -d $TAG || true
-            git -C $repodir tag -a -m "Retag" $TAG
-
-            # Try to build without clearing out build. This allows us to trend toward a
-            # shorter patch set and build time if all goes well ... then fall back to a
-            # hard reset and reconfigure if the fast approach doesn't work.
-            if ! $MAKE -C $repodir/_build dist-gzip distdir=release-patched; then
-                trace "Cleaning out build and trying again"
-
-                rm -rf $repodir/_build
-                autogen_or_configure $repodir
-                $MAKE -C $repodir/_build dist-gzip distdir=release-patched
-            fi
-
-            # TODO: So if the above make dist-gzip fails there is the opportunity
-            # to automatically combine patches at this level, if one patch breaks
-            # building a tarball, but the next fixes it.
-
-            trace "Committing tarball"
-
-            # Now extract the tarball into our stage directory and commit it
-            # HACK: Exclude node_modules/ which was in cockpit tarballs
-            tar -C $stagedir --exclude="node_modules" \
-                --strip-components=1 -xzf - < $repodir/_build/release-patched.tar.gz
-            git -C $stagedir add -f .
-
-            # If nothing added above, then just skip
-            if git -C $stagedir diff-index --exit-code --quiet HEAD --; then
-                continue
-            fi
-
-            # Transfer key information to new stage commit
-            author="$(git -C $repodir show $commit --pretty='%an <%ae>' --no-patch)"
-            date="$(git -C $repodir show $commit --pretty='%ad' --no-patch)"
-            git -C $repodir show $commit --pretty='%B' --no-patch |
-                git -C $stagedir commit -q --author="$author" --date="$date" --file=- --no-edit
+        archive="$(output_tarballs $repodir/_build $SOURCE)"
+    fi
+
+    trace "Committing first tarball"
+    mkdir $stagedir
+    git -C $stagedir init -q
+    git -C $stagedir config core.autocrlf false
+    git -C $stagedir config core.safecrlf false
+    git -C $stagedir config gc.auto 0
+
+    # Mark all sorts of extra files as binary for diffing purposes
+    printf "*.min.* binary\n*.map binary\n/dist/** binary\n" > $stagedir/.git/info/attributes
+
+    # HACK: Exclude node_modules/ which was in cockpit tarballs
+    tar -C $stagedir --exclude="node_modules" \
+        --strip-components=1 -xf "$archive"
```

Comment:
Is this still relevant? I think our node_modules in cockpit tarballs is now a small manageable size, right?

Reply:
It's not, it's still 385 MB. We still have phantomjs in there, as the unit tests still need that (see https://github.com/cockpit-project/cockpit/pull/8314)